core-bundle/src/Resources/contao/library/Contao/Controller.php

```diff
@@ -87,10 +87,35 @@ public static function getTemplate($strTemplate)
 	public static function getTemplateGroup($strPrefix)
 	{
 		$arrTemplates = array();
+		$arrOthers = array();
+
+		$arrMapper = array
```

Comment:
Should this possibly be extendable? Like Isotope could pass an array of keys for Isotope payment steps? Maybe an optional second argument with additional mappers? But then this should look like this:

```
$arrMappers = array
(
    'ce' => &$GLOBALS['TL_CTE'],
    'form' => &$GLOBALS['TL_FFL'],
    'mod' => &$GLOBALS['FE_MOD'],
);
```

Reply:
How would this work? Are you thinking about dispatching an event or about adding the mapper array to `$GLOBALS`?

Reply:
sorry my code example was a bit confusing.

```php
public static function getTemplateGroup($strPrefix, array $arrAdditionalMappers = [])
{
    $arrMappers = array_merge($arrAdditionalMappers, array
    (
        'ce' => array_keys($GLOBALS['TL_CTE']),
        'form' => array_keys($GLOBALS['TL_FFL']),
        'mod' => array_keys($GLOBALS['FE_MOD']),
    ));

    // ...

    if (in_array($strKey, $arrMappings)) { 
        // ...
    }
}
```

Reply:
I see. What about this?

```php
public static function getTemplateGroup($strPrefix, $strMapperKey=null)
{
}

Controller::getTemplateGroup('ce_', 'TL_CTE');
```

Would this be flexible enough or could it be that `isotope_` is a prefix for both content element and module templates?

Reply:
That would require the key to be available as a root element in the global mapper array. I don't want to force that…

Reply:
Agreed. There is a problem with `array_keys()` though, because `FE_MOD` also has groups:

```php
$GLOBALS['FE_MOD'] = array
(
	'navigationMenu' => array
	(
		'navigation'     => 'Contao\ModuleNavigation',
		'customnav'      => 'Contao\ModuleCustomnav',
	// ...
```

Any idea how to solve this smarter than "foreach" and "if is_array"?

Reply:
like `$arrMappers['mod'] = array_keys(array_merge(...$GLOBALS['FE_MOD']));`?

Reply:
Nice, however that does not work with string keys (see https://3v4l.org/4JQ18).

Reply:
This is how it works:

```php
array_merge(...array_values(array_map('array_keys', $GLOBALS['FE_MOD'])))
```

This is a little crazy though, isn't it? 😂

Reply:
`array_keys(array_merge(...array_values($a)));` should do

Reply:
Implemented in f8617c56d2f071eea85387206e82c602403b824f.